MFVNeutralino/plugins/VertexHistos.cc

```diff
@@ -823,81 +1385,359 @@ void MFVVertexHistos::analyze(const edm::Event& event, const edm::EventSetup&) {
-
-    double dbv0     = sv0.rescale_bs2ddist;
-    double dbv1     = sv1.rescale_bs2ddist;
-
-    h_sumdbv->Fill(dbv0 + dbv1, w);
-
-    h_sumdbv_trigcomb->Fill(dbv0 + dbv1 , 0., w);
-    h_caloht_trigcomb->Fill(caloht, 0., w);
-    h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, 0., w);
-
-    if (this_year == 2016) {
-      for (int jt=20; jt<22; jt++){
-        if (mevent->pass_hlt(jt)) {
-          h_sumdbv_trigcomb->Fill(dbv0 + dbv1, float(jt-19)+0.1, w);   //     bin #2 (x-val = 1.1)
-          h_caloht_trigcomb->Fill(caloht, float(jt-19)+0.1, w);                //     bin #2 (x-val = 1.1)
-          h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, float(jt-19)+0.1, w);    // and bin #3 (x-val = 2.1)
-          pass_trig = true;
-          pass_dd   = true;
-        }   
-      }   
-      for (int jt=22; jt<25; jt++){
-        if (mevent->pass_hlt(jt)) {
-          h_sumdbv_trigcomb->Fill(dbv0 + dbv1, float(jt-18)+0.1, w);  // bins #5, 6, 7, (x-vals = 4.1, 5.1, 6.1)
-          h_caloht_trigcomb->Fill(caloht,     float(jt-18)+0.1, w);          
-          h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, float(jt-18)+0.1, w);
-          pass_trig = true;
-          pass_bjet = true;
-        }   
-      }   
-  
-      if (pass_dd)   { h_sumdbv_trigcomb->Fill(dbv0 + dbv1, 3.1, w); h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, 3.1, w); }
-      if (pass_bjet) { h_sumdbv_trigcomb->Fill(dbv0 + dbv1, 7.1, w); h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, 7.1, w); }
-      if (pass_trig) { h_sumdbv_trigcomb->Fill(dbv0 + dbv1, 8.1, w); h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, 8.1, w); }
-    }
-
-    if (this_year == 2017) {
-      for (int jt=18; jt<20; jt++){
-        if (mevent->pass_hlt(jt)) {
-          h_sumdbv_trigcomb->Fill(dbv0 + dbv1, float(jt-17)+0.1, w); 
-          h_caloht_trigcomb->Fill(caloht,      float(jt-17)+0.1, w);          
-          h_dbv0dbv1_trgbit->Fill(dbv0, dbv1,  float(jt-17)+0.1, w);
-          pass_trig = true;
-          pass_dd   = true;
-        }   
-      }   
-      for (int jt=8; jt<10; jt++){
-        if (mevent->pass_hlt(jt)) {
-          h_sumdbv_trigcomb->Fill(dbv0 + dbv1, float(jt-4)+0.1, w); 
-          h_caloht_trigcomb->Fill(caloht,     float(jt-4)+0.1, w);          
-          h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, float(jt-4)+0.1, w);
-          pass_trig = true;
-          pass_bjet = true;
-        }   
-      }   
-  
-      if (pass_dd)   { h_sumdbv_trigcomb->Fill(dbv0 + dbv1, 3.1, w); h_caloht_trigcomb->Fill(caloht, 3.1, w); h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, 3.1, w); }
-      if (pass_bjet) { h_sumdbv_trigcomb->Fill(dbv0 + dbv1, 6.1, w); h_caloht_trigcomb->Fill(caloht, 6.1, w); h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, 6.1, w); }
-      if (pass_trig) { h_sumdbv_trigcomb->Fill(dbv0 + dbv1, 7.1, w); h_caloht_trigcomb->Fill(caloht, 7.1, w); h_dbv0dbv1_trgbit->Fill(dbv0, dbv1, 7.1, w); }
-    }
-
+    h_sum_bsbs2ddist->Fill(mag(sv0.x - mevent->bsx_at_z(sv0.z), sv0.y - mevent->bsy_at_z(sv0.z)) + mag(sv1.x - mevent->bsx_at_z(sv1.z), sv1.y - mevent->bsy_at_z(sv1.z)));
```

Comment:
I think the `this_year` part should go away (pull it from Year.h instead, or something), but are we sure that we can lose the rest with no harm? I'm worried that a lot of this looks like stuff Shaun did a lot on! @pkotamnives @abbywarden @shaunhogan